### ISSUE TITLE
"SYSTEM" -> "Vorschau-Links" broken

### ISSUE THREAD

**Issue by jochi44** [contao/contao#5041]
### Affected version(s)

5.0.0-RC1

### Description

After successfully creating and saving a preview-link in SYSTEM -> Vorschau-Links, accessing the menu item SYSTEM -> Vorschau-Links afterwards leads to an "internal server error".

Details in /var/logs:

```
[2022-07-25T20:12:50.176378+02:00] request.INFO: Matched route "contao_backend". {"route":"contao_backend","route_parameters":{"_route":"contao_backend","_scope":"backend","_token_check":true,"_controller":"Contao\\CoreBundle\\Controller\\BackendController::mainAction"},"request_uri":"https://contao5.xyz.tld/contao?do=preview_link&ref=CxfBwhUv","method":"GET"} []
[2022-07-25T20:12:50.193804+02:00] request.CRITICAL: Uncaught PHP Exception Error: "Cannot create references to/from string offsets" at /var/www/clients/web/contao5/vendor/contao/core-bundle/src/Translation/MessageCatalogue.php line 171 {"exception":"[object] (Error(code: 0): Cannot create references to/from string offsets at /var/www/clients/web/contao5/vendor/contao/core-bundle/src/Translation/MessageCatalogue.php:171)"} []

```

**Comment by fritzmg** [comment]
I can confirm the problem. 

The issue is, that for some reason, `$GLOBALS['TL_LANG']['tl_preview_link']['share']` will, at that point, not be an array containing two elements anymore, but a reference to a string containing the value of `$GLOBALS['TL_LANG']['tl_preview_link']['share'][0]`.

The issue can be fixed by removing the assignment by reference here

https://github.com/contao/contao/blob/eb669536225740719a99927f1518605e4bfe2c48/core-bundle/src/Translation/MessageCatalogue.php#L164

and here

https://github.com/contao/contao/blob/eb669536225740719a99927f1518605e4bfe2c48/core-bundle/src/Translation/MessageCatalogue.php#L171

but I assume the problem lies somewhere else.

**Comment by ausi** [comment]
The `Cannot create references to/from string offsets` error should get fixed by #5058 and #5059

> but I assume the problem lies somewhere else.

I think so too, we should probably investigate where `$GLOBALS['TL_LANG']['tl_preview_link']['share']` get’s modified.

**Comment by ausi** [comment]
> > but I assume the problem lies somewhere else.
> 
> I think so too, we should probably investigate where `$GLOBALS['TL_LANG']['tl_preview_link']['share']` get’s modified.

Done, see #5060
